libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1164,6 +1165,19 @@ blk_hash_t PbftManager::calculateOrderHash(std::vector<DagBlock> const &dag_bloc
   return dev::sha3(order_stream.out());
 }
 
+std::optional<blk_hash_t> findClosestAnchor(const std::vector<blk_hash_t> &ghost, std::vector<blk_hash_t> &dag_order,
+                                            uint32_t included) {
+  for (uint32_t i = included; i > 0; i--) {
+    if (std::count(ghost.begin(), ghost.end(), dag_order[i - 1])) {
+      if (dag_order.size() > i) {
+        dag_order.erase(dag_order.begin() + i, dag_order.end());
+      }
+      return dag_order[i - 1];
+    }
+  }
+  return ghost[1];
```

Comment:
Are you sure this's ghost[1] not ghost[0]? I am not sure if include last finalized DAG anchor in ghost. Please double check
But in case if only has DAG genesis block, that will crash here

Reply:
Yeah. It seems that `ghost[0]` is last finalized DAG anchor. I tried to return `ghost[0]` here and it broke DAG production.
If it will have only genesis block, it won't reach this code because of [pbft_manager.cpp:1237](https://github.com/Taraxa-project/taraxa-node/blob/6059ab3e099d3544ce8f233751963339d772b541/libraries/core_libs/consensus/src/pbft/pbft_manager.cpp#L1237)